lib/db/memory.js

```diff
@@ -187,7 +222,128 @@ MemoryStore.prototype = {
     deleteByUserId(this.tokens, userId);
     deleteByUserId(this.codes, userId);
     return P.resolve();
+  },
+  activateDeveloper: function activateDeveloper(email) {
+    var self = this;
+
+    if (! email) {
+      return P.reject(new Error('Email is required'));
+    }
+
+    return this.getDeveloper(email)
+      .then(function(result) {
+        if (result) {
+          return P.reject(new Error('ER_DUP_ENTRY'));
+        }
+
+        var newId = unique.developerId();
+
+        self.developers[unbuf(newId)] = {
+          developerId: newId,
+          email: email,
+          createdAt: new Date()
+        };
+
+      });
+  },
+  getDeveloper: function getDeveloper(email) {
+    var self = this;
+    var developer = null;
+
+    if (! email) {
+      return P.reject(new Error('Email is required'));
+    }
+
+    Object.keys(self.developers).forEach(function(developerId) {
+      var devEntry = self.developers[developerId];
+
+      if (devEntry.email === email) {
+        developer = devEntry;
+      }
+    });
+
+    return P.resolve(developer);
+  },
+  removeDeveloper: function removeDeveloper(email) {
+    var self = this;
+
+    if (! email) {
+      return P.reject(new Error('Email is required'));
+    }
+
+    Object.keys(self.developers).forEach(function(developerId) {
+      var devEntry = self.developers[developerId];
+
+      if (devEntry.email === email) {
+        delete self.developers[developerId];
+      }
+    });
+
+    return P.resolve();
+  },
+  developerOwnsClient: function devOwnsClient(developerEmail, clientId) {
+    var self = this;
+    var developerId;
+
+    logger.debug('developerOwnsClient');
+    return self.getDeveloper(developerEmail)
+      .then(function (developer) {
+        if (! developer) {
+          return P.reject();
+        }
+        developerId = developer.developerId;
+
+        return self.getClientDevelopers(clientId);
+      })
+      .then(function (developers) {
+        var result;
+
+        function hasDeveloper(developer) {
+          result = developer;
+          return unbuf(developer.developerId) === unbuf(developerId);
+        }
+
+        if (developers.some(hasDeveloper)) {
+          return P.resolve(result);
```

Comment:
This returns the developer object, or a boolean? Looks like the developer object to me, though the name of the method sounds like a question asking for a yes or no.